### PR TITLE
Remove header serialization in favor of setting IIS struct directly.

### PR DESCRIPTION
**Pull request by jkotalik**
Response to https://github.com/aspnet/IISIntegration/pull/465.

### CODE REVIEW

File: src/AspNetCore/Src/managedexports.cxx [hunk 1]
```diff
@@ -355,4 +355,17 @@ http_cancel_io(
...
+HRESULT
+http_response_set_header(
+    _In_ IHttpContext* pHttpContext,
+    _In_ PCSTR pszHeaderName,
+    _In_ PCSTR pszHeaderValue,
+    _In_ USHORT usHeaderValueLength,
+    _In_ BOOL  fReplace
+)
+{
+    return pHttpContext->GetResponse()->SetHeader( pszHeaderName, pszHeaderValue, usHeaderValueLength, fReplace );
```
Comment (Tratcher) [inline src/AspNetCore/Src/managedexports.cxx:368]: There's an indexed overload you can use for known headers so you don't have to serialize and pin their names.
https://msdn.microsoft.com/en-us/library/ms694295(v=vs.90).aspx

Comment (Tratcher) [reply src/AspNetCore/Src/managedexports.cxx:368]: Is http_response_set_header even needed? can't the managed code call SetHeader directly?

Comment (Tratcher) [reply src/AspNetCore/Src/managedexports.cxx:368]: The same for http_set_response_status_code

Comment (jkotalik) [reply src/AspNetCore/Src/managedexports.cxx:368]: The managed code can call SetHeader, but it still needs to do a pinvoke. Either way we need to access the native iis http context. At this point I am keen on not changing it.

Comment (jkotalik) [reply src/AspNetCore/Src/managedexports.cxx:368]: I think we will need to add two native methods to handle this.

Comment (pan-wang) [reply src/AspNetCore/Src/managedexports.cxx:368]: will you set known header and unknown header differently at managed layer? IIS SetHeader api does not distinguish known header and unknown header here. 
internally, IIS SetHeader will calculate the ID of the string input header name to determine it is known one or unknown one. Even given header index, IIS still check whether the header needs to be treated as unknown. I am not sure about the performance difference here. If managed code needs to do some calculation to determine whether the header is known or unknown. You may want to simplify the code to only call IIS SetHeader with string header value.

Comment (Tratcher) [reply src/AspNetCore/Src/managedexports.cxx:368]: The known header API avoids having the managed layer serialize extra strings. As these are the most common headers, that's a lot of allocation, encoding, and pinning we can avoid.

**Review by Tratcher — CHANGES_REQUESTED** [review]
We still don't have a good reason for adding these new methods when the managed layer can p-invoke SetHeader directly. Is there a reason?

**Comment by jkotalik** [comment]
After offline discussion, we are on board with staying with this style. @pan-wang this good to go.

**Review by pan-wang — APPROVED** [review]

**Pull request merged**